### PR TITLE
Adds option to use URL field for teaser card links

### PR DESCRIPTION
**Pull request by kevinhowbrook**
# Description

We would like the option to use URLs in the text only teaser card. If used, they should open links in a new tab.

Link to sample test page: http://localhost:8000/en/youtube/ > Add text only teaser block
Related PRs/issues: #9601 

# Checklist

**Tests**
- [ ] ~Is the code I'm adding covered by tests?~ No

**Changes in Models:**
- [ ] ~Did I update or add new fake data?~ No
- [ ] ~Did I squash my migration?~ I didn't because they are single migration files in 2 apps
- [ ] ~Are my changes [backward-compatible](https://github.com/mozilla/foundation.mozilla.org/blob/main/docs/workflow.md#django-migrations-what-to-do-when-working-on-backward-incompatible-migrations). If not, did I schedule a deploy with the rest of the team?~ I don't think this needs to be backwards compatible as it's a new field on an existing block

**Documentation:**
- [ ] Is my code documented?
- [ ] Did I update the READMEs or wagtail documentation?

### CODE REVIEW

File: network-api/networkapi/wagtailpages/pagemodels/customblocks/text_only_teaser.py [hunk 1]
```diff
@@ -1,13 +1,30 @@
+from django.forms.utils import ErrorList
 from wagtail.core import blocks
+from wagtail.core.blocks.struct_block import StructBlockValidationError
 
 
 class TextOnlyTeaserCard(blocks.StructBlock):
 
     heading = blocks.CharBlock(help_text="Heading for the card.")
     heading_link = blocks.PageChooserBlock(required=False, help_text="Page that the header should link out to.")
```
Comment (tbrlpld) [inline network-api/networkapi/wagtailpages/pagemodels/customblocks/text_only_teaser.py:9]: What do you think about renaming this field to `link_page` or something like that? I think that would feel a bit more consistent from a naming perspective.

**Review by tbrlpld — APPROVED** [review]
@kevinhowbrook This looks good to me. I left one optional question about the other link field name.

File: network-api/networkapi/wagtailpages/pagemodels/customblocks/text_only_teaser.py [hunk 2]
```diff
@@ -1,13 +1,30 @@
...
     body = blocks.TextBlock(help_text="Body text of the card.", required=False, max_length=200)
 
+    def clean(self, value):
+        result = super().clean(value)
+        errors = {}
+
+        if value["heading_link"] and value["link_url"]:
+            errors["heading_link"] = ErrorList(["Please choose between a heading link OR a URL value."])
+        if errors:
+            raise StructBlockValidationError(errors)
```
Comment (tbrlpld) [inline network-api/networkapi/wagtailpages/pagemodels/customblocks/text_only_teaser.py:24]: Can you add some context why are not raising the `StructBlockValidationError` directly when we detect the issue? 


```suggestion
        if value["heading_link"] and value["link_url"]:
            raise StructBlockValidationError(
              {"heading_link": ErrorList(["Please choose between a heading link OR a URL value."])}
            )
```

**Review by tbrlpld — COMMENTED** [review]
Actually just left another question about the structure of the error raising logic.

File: network-api/networkapi/wagtailpages/pagemodels/customblocks/text_only_teaser.py (hunk 1 above)
Comment (kevinhowbrook) [reply network-api/networkapi/wagtailpages/pagemodels/customblocks/text_only_teaser.py]: Sure thing, added in bf60a832c284edc85749b846ba6e33f76d990d29

File: network-api/networkapi/wagtailpages/pagemodels/customblocks/text_only_teaser.py (hunk 2 above)
Comment (kevinhowbrook) [reply network-api/networkapi/wagtailpages/pagemodels/customblocks/text_only_teaser.py]: This is just preference I think, it allows for other errors to be added into the `errors` dict. I'll rework this to be more specific though

Comment (kevinhowbrook) [reply network-api/networkapi/wagtailpages/pagemodels/customblocks/text_only_teaser.py]: Fixed in bdf625d3490912c61bdc08dee10fa43f3d077e9b

**Comment by kevinhowbrook** [comment]
Rebased on main and migrations fixed in e3a222bb71008d6d1d8ee4a7a6bc8c36d01b2690

**Pull request merged**